src/Support/Rules.php

```diff
@@ -44,12 +44,34 @@ public function get(): array
      * @param  array<string,mixed>  $arguments
      * @return array<string,mixed>|string
      */
-    protected function resolveRules($rules, array $arguments)
+    protected function resolveRules($rules, ?string $prefix, array $arguments)
     {
         if (is_callable($rules)) {
             return call_user_func($rules, $arguments, $this->requestArguments);
         }
 
+        if ($prefix && is_array($rules)) {
+            $rules = array_map(
+                function ($rule) use ($prefix) {
+                    if (is_string($rule) && preg_match('/^(required_with(?:out)?(?:_all)?):(.*)$/', $rule, $matches)) {
```

Comment:
Hm, don't we also support the array forms of rules, e.g. `['required_with:…', '…']` or am I on the completely wrong track here?

Yeah, I think tests to play around would help :)

Reply:
Yeah `['required_with' => ['thing']]` might be a thing

Reply:
Also this is apparently a thing

```php
                'rules' => function ($rules) {
                    return ['required'];
                },
```

Not sure how viable this approach is.